src/deployment/mod.rs

```diff
@@ -0,0 +1,1086 @@
+            Some(ref deployer) => {
+                let deployer = match chain_config.accounts.get(deployer) {
+                    Some(deployer) => deployer,
+                    None => {
+                        return Err(format!(
+                            "{} unable to retrieve account '{}'",
+                            red!("x"),
+                            deployer
+                        ));
+                    }
+                };
+                deployer
+            }
+            None => default_deployer,
+        };
+
+        let emulated_sender = match PrincipalData::parse_standard_principal(&deployer.address) {
+            Ok(res) => res,
+            Err(_) => {
+                return Err(format!(
+                    "unable to turn emulated_sender {} as a valid Stacks address",
+                    deployer.address
+                ))
+            }
+        };
+
+        let mut path = manifest.get_project_root_dir();
+        path.push(&config.path);
+        let source = match std::fs::read_to_string(&path) {
+            Ok(code) => code,
+            Err(err) => {
+                return Err(format!(
+                    "unable to read contract at path {:?}: {}",
+                    config.path, err
+                ))
+            }
+        };
+
+        let contract = EmulatedContractPublishSpecification {
+            contract_name,
+            emulated_sender,
+            source,
+            relative_path: config.path.clone(),
+        };
+
+        let contract_id = QualifiedContractIdentifier::new(
+            contract.emulated_sender.clone(),
+            contract.contract_name.clone(),
+        );
+
+        contracts.insert(contract_id, contract);
+    }
+
+    let session = Session::new(settings);
+
+    let mut contract_asts = HashMap::new();
+    let mut contract_diags = HashMap::new();
+
+    for (contract_id, contract) in contracts.iter() {
+        let (ast, diags, _) = session.interpreter.build_ast(
+            contract_id.clone(),
+            contract.source.clone(),
+            parser_version,
+        );
+        contract_asts.insert(contract_id.clone(), ast);
+        contract_diags.insert(contract_id.clone(), diags);
+    }
+
+    let dependencies =
+        ASTDependencyDetector::detect_dependencies(&contract_asts, &requirements_asts);
+
+    let mut dependencies = match dependencies {
+        Ok(dependencies) => dependencies,
+        Err(_) => {
+            return Err(format!("unable to detect dependencies"));
+        }
+    };
+
+    for contract_id in boot_contracts_ids.into_iter() {
+        dependencies.insert(contract_id.clone(), DependencySet::new());
+    }
+
+    dependencies.extend(requirements_deps);
+
+    let ordered_contracts_ids = match ASTDependencyDetector::order_contracts(&dependencies) {
+        Ok(ordered_contracts_ids) => ordered_contracts_ids
+            .into_iter()
+            .map(|c| c.clone())
+            .collect::<Vec<_>>(),
```

Comment:
I don't think this cloning is necessary. Can you just use `ordered_contracts_ids`? When the contract ids are used later, they are either used as references or they are cloned, so this clone should be extra.

Reply:
Addressed with latest

Reply:
ha yes, my bad.

Reply:
Addressed with https://github.com/hirosystems/clarinet/pull/333/commits/c03549d1fa3804faa9fbd19f01e0933d0051ae0f